tensorflow_encrypted/protocol/pond.py

```diff
@@ -1572,10 +1595,72 @@ def _transpose_masked(prot, x_masked):
     x_t = PondMaskedTensor(prot, x_unmasked_t, a_t, a0_t, a1_t, alpha_on_0_t, alpha_on_1_t)
     return x_t
 
+#
+# strided slice helpers
+#
+
+
+def _strided_slice_public(prot, x, args, kwargs):
+    assert isinstance(x, PondPublicTensor)
+
+    x_on_0, x_on_1 = x.unwrapped
+
+    with tf.name_scope('strided_slice'):
+
+        with tf.device(prot.server_0.device_name):
+            x_on_0_t = x_on_0.strided_slice(args, kwargs)
+
+        with tf.device(prot.server_1.device_name):
+            x_on_1_t = x_on_1.strided_slice(args, kwargs)
+
+    x_t = PondPublicTensor(prot, x_on_0_t, x_on_1_t)
+    return x_t
+
+
+def _strided_slice_private(prot, x, args, kwargs):
+    assert isinstance(x, PondPrivateTensor)
+
+    x0, x1 = x.unwrapped
+
+    with tf.name_scope('strided_slice'):
+
+        with tf.device(prot.server_0.device_name):
+            x0_t = x0.strided_slice(args, kwargs)
+
+        with tf.device(prot.server_1.device_name):
+            x1_t = x1.strided_slice(args, kwargs)
+
+    x_t = PondPrivateTensor(prot, x0_t, x1_t)
+    return x_t
+
+
+def _strided_slice_masked(prot, x_masked, args, kwargs):
+    assert isinstance(x_masked, PondMaskedTensor)
+
+    a, a0, a1, alpha_on_0, alpha_on_1 = x_masked.unwrapped
+
+    with tf.name_scope('strided_slice'):
+
+        with tf.device(prot.crypto_producer.device_name):
+            a_t = a.strided_slice()
```

Comment:
you're not passing `args` and `kwargs` in neither of the three applications here

Reply:
Code catch 😄